neo4j/internal/bolt/bolt5.go

```diff
@@ -274,6 +290,27 @@ func (b *bolt5) Connect(ctx context.Context, minor int, auth map[string]any, use
 	return nil
 }
 
+func (b *bolt5) ChangeUser(ctx context.Context, auth map[string]any) error {
```

Comment:
I think it makes sense to export that function, but it may be too early yet, since it's not used anywhere else?
Once we need to expose that at the driver/session level, `ChangeUser` will probably need to be defined at the `Connection` interface level as well